kopf/_core/engines/daemons.py

```diff
@@ -420,6 +420,21 @@ async def _runner(
         if stopper.reason is None:
             memory.forever_stopped.add(handler.id)
 
+        # If this daemon is never going to be called again, we can release the
+        # live_fresh_body to save some memory.
+        if handler.id in memory.forever_stopped:
+            # If any other running daemon is referencing this Kubernetes 
+            # resource, we can't free it
+            can_free = True
+            this_daemon = daemons[handler.id]
+            for running_daemon in memory.running_daemons.values():
+                if running_daemon is not this_daemon:
+                    if running_daemon.handler.selector.check(cause.resource):
```

Comment:
running_daemon is the other daemon.  If running_daemon and this_daemon are referencing the same Kubernetes object, then we can't free the body since it's still in-use by the other daemon.

Reply:
Thanks for the clarification. I think this 2nd "if" is not needed. 

The cross-object isolation happens much earlier, a few architectural levels above, i.e. when the individual objects are split into separate processing streams:

* First, in `watcher()`/`worker()` using the `(resource, uid)` as the key (also see `get_uid()` there).
* Second, where the singular `memory` is extracted from the plural `memories` via `memories.recall()`.

When it comes to the `memory` & `memory.running_daemons`, it is already dedicated to a specific object. All other objects (of the same and other kinds) have their own `memory` struct.

So, to my understanding, the check for _"other daemons except for the current one"_ is sufficient (which is done on the line above). The selector check is redundant — it will always be `True` because of such a split.

So, this 2nd "if" should be removed. Otherwise, the PR is clear, I am ready to merge and release it ASAP.

Reply:
if removed, new push coming shortly.